lib/Doctrine/ODM/MongoDB/Mapping/MappingException.php

```diff
@@ -276,4 +278,39 @@ public static function schemaValidationError(int $errorCode, string $errorMessag
     {
         return new self(sprintf('The following schema validation error occurred while parsing the "%s" property of the "%s" class: "%s" (code %s).', $property, $className, $errorMessage, $errorCode));
     }
+
+    public static function enumsRequirePhp81(string $className, string $fieldName): self
+    {
+        return new self(sprintf('Enum types require PHP 8.1 in %s::$%s', $className, $fieldName));
+    }
+
+    public static function nonEnumTypeMapped(string $className, string $fieldName, string $enumType): self
+    {
+        return new self(sprintf(
+            'Attempting to map non-enum type %s as enum in document class %s::$%s',
```

Comment:
```suggestion
            'Attempting to map a non-enum type %s as an enum: %s::%s',
```

This should be the most consistent with other exceptions we have